relayer.go

```diff
@@ -86,6 +93,65 @@ func EncodeTransactionsForRelaying(relayer Relayer, walletConfig WalletConfig, w
 }
 
 func WaitForMetaTxn(ctx context.Context, provider *ethrpc.Provider, metaTxnID MetaTxnID, timeout time.Duration) (*types.Receipt, error) {
-	// TODO.
-	return nil, fmt.Errorf("TODO")
+	timeoutTime := time.Now().Add(timeout)
+
+	// Start listening logs from current block - 1024
+	block, err := provider.BlockNumber(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	lastBlockNumber := block - 1024
+
+	metaTxIdBytes := common.Hex2Bytes(string(metaTxnID))
+
+	// All transactions must change nonces
+	// so load all nonce changes and search the logs
+	nonceChangedTopics := [][]common.Hash{{nonceChangedTopic}}
+
+	// Load all logs until we found the receipt or we reach timeout
+	for time.Now().Before(timeoutTime) {
+		block, err := provider.BlockNumber(ctx)
+
+		if err != nil {
+			time.Sleep(time.Second)
+			continue
+		}
+
+		query := ethereum.FilterQuery{
+			FromBlock: big.NewInt(int64(lastBlockNumber) - 12),
```

Comment:
Think we shouldn't hardcode the 12 value and set it as a variable like in sequence.js.

Reply:
I agree, I will leave a comment so we later add some sort of configuration parameter or something.